TESTING.md

```diff
@@ -2,12 +2,124 @@
+* (2) An organization
+    * (3) An organization administrator user
+    * (4) A Virtual Data Center (VDC) with enough resources to build several vApps and VMs, and preferably 2 storage profiles
+        * (5) an Edge gateway with a default gateway
+    * (6) A catalog
+    * (7) A vApp template in the catalog
+
+Optionally, to run tests in [go-vcloud-director](https://github.com/vmware/go-vcloud-director), you will also need one or
+two Org VDC networks and a media item.
+
+
+This is how you can build the environment using Terraform itself.
+
+The above entities are defined in the test configuration file (see `sample_vcd_test_config.json` for the full list of
+entities to provide). Here is an excerpt of where these names come from:
+
+```
+{
+  "vcd": {
+     "org": "datacloud",                    // (2)
+     "vdc": "datacloudvdc",                 // (4)
+     "catalog": {
+       "name": "datacloudcat",              // (6)
+       "catalogItem": "photon-hw11"         // (7)
+     },
+     [...]
+   },
+   "networking": {
+     "edgeGateway": "gwdatacloud",           // (5)
+     "externalNetwork": "extnetdatacloud",   // (1)
+     [...]
+     }
+   },
+   [...]
+ }
+
+```
+
+The vCD environment builder uses the test system information to create these entities. When triggered, it will take the
+names of the entities and some of the configuration parameters from the regular test configuration file. It will then
+integrate such data with additional information that is only used for the environment build, and it's listed under the
+`testEnvBuild` section in the file:
+
+```
+{
+   [...]
+  "testEnvBuild": {
+    // Indispensable data, which is not normally used in the tests but is mandatory to create the
+    // environment
+
+    "gateway": "GATEWAY_IP",
+    "netmask": "255.255.224.0",
+    "externalNetworkStartIp": "STARTING_IP",
+    "externalNetworkEndIp": "END_IP",
+    "dns1": "DNS_IP",
+
+    // Optional data. If not provided, the corresponding values
+    // from the rest of the configuration file will be used
+
+    "dns2": "8.8.4.4",
+    "externalNetworkPortGroup": "VM Network",
+    "externalNetworkPortGroupType": "NETWORK",
+    "ovaPath": "/path/to/photon-hw11-3.0-26156e2.ova",
+    "media_path":  "/path/to/test.iso",
+
+    // extra data. If not provided, these resources will not be created
+    // These items can be used for go-vcloud-director testing
+
+    "mediaName": "test_media",
+    "storageProfile2": "Development",
+    "routedNetwork": "net_datacloud_r",
+    "isolatedNetwork": "net_datacloud_i",
+    "directNetwork": "net_datacloud_d"
+  }
+}
+```
+
+When this section is filled, the test system has all the information to create all the elements (except, for now, the
+org admin user). When you run the command `make test-binary-prepare`, among the files ready to run there will be one named
+`cust.full-env.tf`, containing all the information to populate your vCD with the test resources.
```

Comment:
I suggest pushing this paragraph about `cust.full-env.tf` down below the `make test-env-build` introduction, so for the beginner it's clear he only needs test-env-build, but then if there are issues or he wants to see the config, he can use the advanced flow.

Reply:
I believe this section is right where it belongs, as it mentions the section, and it would be more difficult to understand if we put it far away.